tools/src/wyvern/tools/typedAST/core/expressions/Application.java

```diff
@@ -141,22 +160,31 @@ public FileLocation getLocation() {
 	public Expression generateIL(GenContext ctx, ValueType expectedType) {
 		CallableExprGenerator exprGen = function.getCallableExpr(ctx);
 		List<FormalArg> formals = exprGen.getExpectedArgTypes(ctx);
-		
+
+        int offset = 0;
 		// generate arguments		
 		List<Expression> args = new LinkedList<Expression>();
+        if(!this.generics.isEmpty()) {
+            for(String generic : this.generics) {
+                 args.add(new wyvern.target.corewyvernIL.expression.New(new TypeDeclaration(DefDeclaration.GENERIC_MEMBER, new NominalType("", generic), this.location)));
```

Comment:
Instead of GENERIC_MEMBER, take formals.get(i) and extract the name, then remove the GENERIC_PREFIX prefix from it.  That gives you the name of the type member.

Instead of new NominalType("", generic), you should follow the code in UnresolvedType.getILType() to generate the ValueType properly.